ui/src/main/Footer/Footer.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import versionService from "../../services/VersionService/VersionService";
+import Spinner from "react-bootstrap/Spinner";
+import { usePromise } from "react-use-promise-matcher";
+import { BsExclamationCircle } from "react-icons/bs";
+
+interface VersionData {
+  buildDate: string;
+  buildSha: string;
+}
+
+const Footer: React.FC = () => {
+  const [result, load] = usePromise<VersionData, any, any>(() => versionService.getVersion());
+
+  React.useEffect(() => {
+    load();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
In order to get rid of this es-lint-disable you can extract the inline callback from `usePromise` to a function declared outside of the component, and then use this function inside `usePromise` and then use the `load` function in the dependencies array e.g.
`const getVersion = () => versionService.getVersion();
 const Footer: React.FC = () => {
  const [result, load] = usePromise<VersionData, any, any>(getVersion);
React.useEffect(() => {
    load();
}, [load]);
(...)
`

Reply:
`versionService.getVersion()` is already a function declared outside component in separate module. `load` changes reference anyway and causes rerender.